File: Prácticas/P4/RUBY/Civitas/lib/sorpresa_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

require_relative "tablero.rb"
require_relative "diario.rb"
require_relative "jugador.rb"
require_relative "mazo_sorpresas.rb"

module Civitas
  class SorpresaPorCasaHotel < Sorpresa
    def initialize(tablero,valor,texto)
      @tablero = tablero
      @valor = valor
      @texto = texto
    end
    
    def aplicar_a_jugador(actual,todos)
      if(jugador_correcto(actual,todos))
        informe(actual,todos)
        num_casas_hoteles = todos[actual].cantidad_casas_hoteles
        importe = @valor*num_casas_hoteles
        todos[actual].modificar_saldo(importe)
      end
    end
    
    def to_s
      to_s="Sorpresa por casa hotel {#{@texto}, valor=#{@valor}}"
    end
  end
end

File: Prácticas/P4/JAVA/Civitas/src/civitas/SorpresaPorCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author Jose Padial
 */
public class SorpresaPorCasaHotel extends Sorpresa{
    private String texto;
    private int valor;
    private Tablero tablero;
    
    SorpresaPorCasaHotel(Tablero tablero, int valor, String texto){
        this.tablero = tablero;
        this.valor = valor;
        this.texto = texto;
    }
    
    @Override
    void aplicarJugador(int actual, ArrayList<Jugador> todos){
        if(todos.get(actual) != null){
            informe(actual,todos);
            int numCasasHoteles = todos.get(actual).cantidadCasasHoteles();
            int importe = valor * numCasasHoteles;
            todos.get(actual).modificarSaldo(importe);
        }
    }

    @Override
    public String toString() {
        return "SorpresaPorCasaHotel{" + "texto=" + texto + ", valor=" + valor + ", tablero=" + tablero + '}';
    } 
}
